### ISSUE TITLE
SmtpClient is not being correctly disposed of

### ISSUE THREAD

**Issue by bartuszekj** [elmah/Elmah#399]
SmtpClient implements IDisposable as of .NET 4.0. ErrorMailModule.SendMail method does not correctly dispose of its SmtpClient. This causes the smtp connection to hang for an extended amount of time. Added benefit of properly closing/disposing of the SmtpClient instance is that an smtp QUIT command can be properly sent to the server.

To maintain compatibility between .NET versions, I suggest implementing the following solution:

```c#
var client = new SmtpClient();

using (client as IDisposable) // allow client not to implement IDisposable, but correctly handle the 4.0+ implementation
{
   // ... do all the SmtpClient work
   client.Send(mail);
}

**Comment by atifaziz** [comment]
@bartuszekj Thanks for bringing this up! Do you want to try and submit a PR against the [1x branch][1x] with the compatible change you proposed?

[1x]: https://github.com/elmah/Elmah/tree/1x

**Comment by bartuszekj** [comment]
Done. It did create a new issue for this, though (#401).

**Comment by atifaziz** [comment]
[Merged][m] & closed by 6861de61ddf80000e7204cd976418bd771921ccc

Thanks! Should be in the next service pack release.

[m]: https://github.com/elmah/Elmah/tree/6861de61ddf80000e7204cd976418bd771921ccc
